### PR TITLE
Handle transformation and solr indexing errors

### PR DESCRIPTION
**Pull request by eliotjordan**
Work towards #268

- Index documents one at a time after batch Solr update returns an error
- Assign default value when title is missing
- Resolves a couple of test warnings.

Current counts:
Figgy: 56172
Hydration Cache: 56106
Transformation Cache: 56106
Solr Records: 56106

### CODE REVIEW

File: lib/dpul_collections/solr.ex [hunk 1]
```diff
@@ -96,11 +98,35 @@ defmodule DpulCollections.Solr do
...
+
+  def add(docs, collection) when length(docs) > 1 do
+    response =
+      Req.post!(
+        update_url(collection),
+        json: docs
+      )
+
+    if response.status != 200 do
+      Enum.each(docs, fn doc -> add([doc]) end)
```
Comment (tpendragon) [inline lib/dpul_collections/solr.ex:111]: Suggestion: `docs |> Enum.each(&add/1)` or `Enum.each(docs, &add/1)`

File: lib/dpul_collections/indexing_pipeline/figgy/hydration_cache_entry.ex [hunk 2]
```diff
@@ -27,7 +27,7 @@ defmodule DpulCollections.IndexingPipeline.Figgy.HydrationCacheEntry do
 
     %{
       id: id,
-      title_txtm: get_in(metadata, ["title"]),
+      title_txtm: extract_title(metadata),
```
Comment (tpendragon) [inline lib/dpul_collections/indexing_pipeline/figgy/hydration_cache_entry.ex:30]: I like that we're starting to get to a pattern here.

**Review by tpendragon — APPROVED** [review]
Just a small style suggestion, but no blockers.

Looks great!

File: lib/dpul_collections/solr.ex (hunk 1 above)
Comment (eliotjordan) [reply lib/dpul_collections/solr.ex]: Had to change the guard clauses a bit, but overall, I think it's better now.

**Review by tpendragon — APPROVED** [review]

**Pull request merged**